source/Panel.cpp

```diff
@@ -101,9 +101,15 @@ void Panel::AddZone(const Rectangle &rect, const function<void()> &fun)
 
 
 
-void Panel::AddZone(const Rectangle &rect, SDL_Keycode key)
+void Panel::AddZone(const Rectangle &rect, SDL_Keycode key, const vector<string> &conditions)
```

Comment:
The `conditions` parameter seems to list the conditions that need to be true for the UI-element to respond to input. I feel that the variable needs a more clear name if we want to go this route (something like `conditionsToEnable`).

I'm also not sure on why we couldn't just use the lambda functions as before? Why do we need this new variant that internally just calls the lambda variant of this function? (Is it because the conditions cannot be captured in the lambda function if we don't have this unique function?)

Reply:
> I'm also not sure on why we couldn't just use the lambda functions as before? Why do we need this new variant that internally just calls the lambda variant of this function? (Is it because the conditions cannot be captured in the lambda function if we don't have this unique function?)

We are still using a lambda function as before; I simply added a condition check inside it.

There is no really easy way of doing it from Interface; it doesn't have access to KeyDown, so it can't construct the lambda, and it also requires the Information instance.

The Information cannot be passed as a parameter to the lambda as it is generally created in the Draw() call, not where mouse events are handled. This can be worked around by simply caching this Information state in a member variable, and accessing that from the lambda.

Reply:
I think this is clear now after the rename.